### ISSUE TITLE
Fixer service starts the maintenance mode every five minutes, but doesn't do anything

### ISSUE THREAD

**Issue by Pajkastare** [nextcloud-snap/nextcloud-snap#2763]
#### Describe the bug
The fixer service (the `systemd` service `snap.nextcloud.nextcloud-fixer.service`, which calls `/usr/bin/snap run nextcloud.nextcloud-fixer` when triggered) consumes too much resources when it runs every five minutes, but worst of all is that it starts the maintenance mode. This is very disruptive to actually using Nextcloud.

I'm no snap expert, but it seems to me that the problem (the entering of maintenance mode automatically) is because this directory:
`/snap/nextcloud/42571/fixes/existing-install/maintenance`
...Contains `1_convert-filecache-bigint.sh`, which in turn contains the line `occ -n db:convert-filecache-bigint`.

**Why would this maintenance-mode-requiring script have to be called every five minutes?**
I just refreshed the snap before making this bug report, so I should be on the latest stable version.

#### To Reproduce
Just watch `journalctl -f` when the fixer service is running

#### Expected behavior
A five-minute periodic "fixer" job should not start the maintenance mode every single time.

#### Screenshots
N/A

#### OS/snapd/snap version
Ubuntu 20.04.6 LTS on a Raspberry Pi 4 with 4 GB RAM.

    $ snap list nextcloud
Name       Version      Rev    Tracking       Publisher   Notes
nextcloud  28.0.6snap1  42571  latest/stable  nextcloud✓  -

    $ snap version
snap    2.63
snapd   2.63
series  16
ubuntu  20.04
kernel  5.4.0-1101-raspi

#### Logs
As mentioned above, the fixer service (which runs every 5 minutes) starts the maintenance mode every time. 

This is a cleaned-up relevant part shown by `journalctl -f`, the "All tables already up to date!" is printed to the console when running the `nextcloud.occ -n db:convert-filecache-bigint` command:

> nextcloud.nextcloud-fixer[2786257]: run-parts: executing /snap/nextcloud/42571/fixes/existing-install/7_add_maintenance_window.sh
> nextcloud.nextcloud-fixer[2787374]: 1
> nextcloud.nextcloud-fixer[2786257]: run-parts: executing /snap/nextcloud/42571/fixes/existing-install/99_notify_admins.sh
> nextcloud.nextcloud-fixer[2785869]: **Enabling maintenance mode...**
> nextcloud.nextcloud-fixer[2785869]: done
> nextcloud.nextcloud-fixer[2787617]: run-parts: executing /snap/nextcloud/42571/fixes/existing-install/maintenance/1_convert-filecache-bigint.sh
> nextcloud.nextcloud-fixer[2787683]: Nextcloud is in maintenance mode, no apps are loaded.
> nextcloud.nextcloud-fixer[2787683]: Commands provided by apps are unavailable.
> nextcloud.nextcloud-fixer[2787683]: All tables already up to date!
> nextcloud.nextcloud-fixer[2785869]: **Disabling maintenance mode...**
> nextcloud.nextcloud-fixer[2785869]: done
> systemd[1]: snap.nextcloud.nextcloud-fixer.service: Succeeded.

#### Regarding workarounds and fixes
Since the snap is a read-only filesystem, I cannot "just" delete the `1_convert-filecache-bigint.sh` file. I could change the `systemd` service file no never run the fixer service at all, which is probably not great for stability (or run it just once a day via a manual cron job or custom `systemd` timer), but I should not have to. I think I recall that the Nextcloud "cron" interval could be changed using `nextcloud.occ`, but I don't remember the syntax right now.

At the very least, the `nextcloud.occ -n db:convert-filecache-bigint` command should not be run without first confirming that there is an actual need to run it (in maintenance mode). But that is probably nothing I can affect if I just use the snap.

**Comment by scubamuc** [comment]
@Pajkastare welcome and thanks for your request

please run the [debugging script](https://github.com/nextcloud-snap/nextcloud-snap/wiki/Issue-Debugging-Scripts#debugging-script) and post results here

**Comment by Pajkastare** [comment]
> @Pajkastare just found this [#2349 (comment)](https://github.com/nextcloud-snap/nextcloud-snap/issues/2349#issuecomment-1465151069) in one of your older issues, did you disable preview generator as suggested? assuming this is an upgrade issue as before.

I did not do anything to `previewgenerator` prior to doing the upgrade (only `sudo snap refresh nextcloud`, no other preparation). Nevertheless, I did disable it (`sudo nextcloud.occ app:disable previewgenerator`), but the maintenance mode problem persisted.


> @Pajkastare welcome and thanks for your request
> 
> please run the [debugging script](https://github.com/nextcloud-snap/nextcloud-snap/wiki/Issue-Debugging-Scripts#debugging-script) and post results here
> 
> > I think I recall that the Nextcloud "cron" interval could be changed using nextcloud.occ, but I don't remember the syntax right now.
> 
> see [managing cron](https://github.com/nextcloud-snap/nextcloud-snap/wiki/Managing-cron)

Thanks for the links - Regarding cron, I had apparently alreday set it to 24h (`sudo snap get nextcloud nextcloud.cron-interval`), but the fixer script runs every five minutes. This got me digging, since it was obviously something other than the cron service that started the fixer script all the time.


It appeared it was another cron job recently created by myself, which started/stopped nextcloud if the external storage (which I access via Samba and mount under `/mnt/`, since the disk performance on the Raspberry Pi was too poor for running Nextcloud well enough) was available/unavailable, and that cron job called "snap start nextcloud" even if it was already running. I fixed it to do nothing if Nextcloud was in fact running, and the maintenance mode problems went away.


So, **today I learned** that the fixer service is called every time the Nextcloud snap starts, and that running `snap start $NAME_OF_SNAP` runs all startup code for that snap even if it was already running. 


In other words, let's close this issue, it was my own fault, thanks for the help and sorry for wasting your time.

**Comment by Pajkastare** [comment]
Problem not related to the Nextcloud snap. (Erroneously believed that calling "snap start nextcloud" would not do anything if Nextcloud was already running, but the fixer service will be executed anyway.)

**Comment by Pajkastare** [comment]
Problem not related to the Nextcloud snap. (Erroneously believed that calling "snap start nextcloud" would not do anything if Nextcloud was already running, but the fixer service will be executed anyway.)
